src/entity/ReputationEvent.ts

```diff
@@ -23,22 +23,23 @@ export enum ReputationType {
 }
 
 const reputationReasonAmount: Record<ReputationReason, number> = {
-  [ReputationReason.PostUpvote]: 10,
+  [ReputationReason.PostUpvoted]: 10,
   [ReputationReason.PostBanned]: -100,
-  [ReputationReason.CommentUpvote]: 50,
+  [ReputationReason.CommentUpvoted]: 50,
   [ReputationReason.PostReportConfirmed]: 100,
   [ReputationReason.SourceRequestApproved]: 200,
 };
 
+export const REPUTATION_THRESHOLD = parseInt(
+  process.env.REPUTATION_THRESHOLD || '250',
+);
+
 @Entity()
 export class ReputationEvent {
-  @PrimaryColumn({ default: () => 'now()' })
-  timestamp: Date;
-
-  @PrimaryColumn({ length: 36, default: '' })
-  grantById: string;
+  @Index({ unique: true })
```

Comment:
I mean primary constraint is just a combination of `unique` and `not null` - so are we not gonna make this a primary column anymore since it might limit the capacity of the table?

Let me revert this back to default an empty string as discussed.

Reply:
@idoshamun just wanted to confirm if we would still push with `PrimaryColumn` for the `grantById` since under the hood it is just a combination of a `unique` constraint with `not null` that might limit the table.

Reply:
Composite primary keys like we have here force uniqueness on the combination, not on every column that makes up the composite key. In our case this is exactly what we want

Reply:
Ah, cool, I think that could also be the case for the `unique` constraint why it has allowed multiple `null` values initially. Let me apply the `PrimaryColumn` and the supposed empty string.

Reply:
Please also check the other comments I submitted outside the review